src/com/facebook/buck/json/buck.py

```diff
@@ -201,9 +204,39 @@ def add_rule(rule, build_env):
             # the user to re-run `buck autodeps` in this scenario. Unfortunately, we do not have
             # a mechanism to relay warnings from buck.py at the time of this writing.
             pass
+    if rule['buck.type'] in ('apple_library', 'apple_binary'):
+        filter_and_generate_swift_rule(rule, build_env)
     build_env.rules[rule_name] = rule
 
 
+def filter_and_generate_swift_rule(rule, build_env):
```

Comment:
If we are going to create the `swift_library` companion rule at the transformation phrase, we will have to populate manually its constructor arguments from its implicit parent (`apple_library` or `apple_binary`), create artificial build target & flavors, etc... This task is pretty much similar to parsing, not to mention that we will also have to change the constructor args of the parent (for removing swift sources), which is supposed to be immutable at this phrase. 
I spent sometimes today trying to do it properly, but the implementation doesn't look pretty, if you guys have a way to start with, I would love to know. Otherwise, I will take the alternative way.